graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java

```diff
@@ -107,10 +112,14 @@ public Response update(@ApiParam(name = "configClass", value = "The name of the
             throw new NotFoundException("Couldn't find configuration class \"" + configClass + "\"");
         }
 
-        final Object o = objectMapper.readValue(body, cls);
-        clusterConfigService.write(o);
-
-        return Response.accepted(o).build();
+        try {
+            final Object o = objectMapper.readValue(body, cls);
+            clusterConfigService.write(o);
+            return Response.accepted(o).build();
+        }catch(Exception e) {
+            LOG.error("Could not write new cluster config.", e);
+            return Response.status(400).build();
```

Comment:
Maybe simply throw a `WebApplicationException` here and add the original exception for its stack trace.